### ISSUE TITLE
Septerra Core v1.04.0.3 don't play movies (Wine and Linux)

### ISSUE THREAD

**Issue by TaxAkla** [M-HT/SR#8]
First, thanks for your good work. Having Septerra Core running natively on Linux is fantastic.

My OS is Linux Mint 19.3 64bits. The game is from GOG.

I followed the instructions and installed the 32-bit libraries, them I started the game through Septerra.sh binary. The game run but without playing any movies.

Then I ran through the terminal and got this errors.

[core] Error: Opening failed (unsupported filetype)
[core] Error: Opening failed (unsupported filetype)
[core] Error: Opening failed (unsupported filetype)
[core] Error: Opening failed (unsupported filetype)

***
My next test was under Wine (Lutris 5-5 fshack) I know that isn't supported but I like to play GOG games through GOG Galaxy thanks to Lutris. Under this bottle the game has crackling sounds and don't play movies, too. Another issue is the game don't go fullscreen even when I change that option on the CFG.
I don't know how to get error report on Windows.

**Comment by M-HT** [comment]
I downloaded and installed Linux Mint 19.3 64bits Cinnamon edition. I upgraded packages and installed packages from readme (libsdl2-2.0-0:i386 libmpg123-0:i386 libquicktime2:i386). I installed Septerra Core from GOG (linux version). When I run the game (using Septerra.sh) the movies are playing. I don't know what you did differently.
___
As for running under Wine, I don't see the point when there is a working Linux version.

**Comment by TaxAkla** [comment]
My Septerra installation is the windows version on a wine bottle. I just pasted your Linux build and tried run. Did your test was with the Linux installer? I will download the Linux one and try again.

**Comment by TaxAkla** [comment]
Found that SR uses something from the Linux installation that don't exist on the Windows/Wine one. The movies are working on the Linux and aren't on the Windows folder.

*********
The Linux folder with working movies gives this messages.

[svq1 @ 0x912e9e0] embedded message:
MSTBPGM240
[svq1 @ 0x91efd60] embedded message:
TGCSMDT240
[svq1 @ 0x921d500] embedded message:
IHWGAPV240
[svq1 @ 0x92503c0] embedded message:
IHTCTK1240
[svq1 @ 0xa739780] embedded message:
BKWSTH2192
---------------------------

Can I help with more information?

**Comment by M-HT** [comment]
I used GOG Windows installer (setup_septerra_core_2.0.0.10.exe) and the movies are playing. What installer are you using ?

**Comment by TaxAkla** [comment]
The same installer here.

Bellow are the versions of my libraries. Something wrong?

libquicktime2:i386 is already the newest version (2:1.2.4-11build1).
libmpg123-0:i386 is already the newest version (1.25.10-1).
libsdl2-2.0-0:i386 is already the newest version (2.0.10+dfsg1-2ubuntu5~ubuntu18.04.1~ppa1)

**Comment by M-HT** [comment]
My libsdl2-2.0-0:i386 has version 2.0.8+dfsg1-1ubuntu1.18.04.4. I don't think it's a problem, but if you tell me which ppa you're using I can test it.

Can you give me a list of files (using `ls -og`) from the windows version (where movies don't work) ?

This is a list from my version (after installation):

    $ ls -og
    total 573808
    -rw-r--r-- 1  74477608 Oct 15  1999  anim.db
    -rw-r--r-- 1      7448 Apr 11 12:30  EULA.txt
    -rw-r--r-- 1     96264 Sep 14  2012  GameuxInstallHelper.dll
    -rw-r--r-- 1    114629 Jul 29  2013  gfw_high.ico
    -rw-r--r-- 1    255296 Jan 24  2014  goggame.dll
    -rw-r--r-- 1     69248 Sep 14  2012  gog.ico
    -rw-r--r-- 1      1449 Apr 11 12:30 'Launch Septerra Core.lnk'
    -rw-r--r-- 1 226736914 Oct 15  1999  level.db
    -rw-r--r-- 1  19476566 Oct 15  1999  M1.DB
    -rw-r--r-- 1  13978922 Oct 15  1999  M2.DB
    -rw-r--r-- 1   3369686 Oct 15  1999  M3.DB
    -rw-r--r-- 1   5550734 Oct 15  1999  M4.DB
    -rw-r--r-- 1   5850614 Oct 15  1999  M5.DB
    -rw-r--r-- 1   2536074 Oct 15  1999  M6.DB
    -rw-r--r-- 1   3200730 Oct 15  1999  M7.DB
    -rw-r--r-- 1   7421734 Oct 15  1999  M8.DB
    -rw-r--r-- 1  14202626 Oct 15  1999  M9A.DB
    -rw-r--r-- 1  14218506 Oct 15  1999  M9B.DB
    -rw-r--r-- 1   1658551 Mar 18  2002  Manual.pdf
    -rw-r--r-- 1   2502586 Oct 15  1999  MONOLITH.DB
    -rw-r--r-- 1   6676480 Sep 23  2004  QuickTime.qts
    -rw-r--r-- 1     35824 Mar 30  2009  README.TXT
    -rw-r--r-- 1    939075 Oct 15  1999  septerra.db
    -rw-r--r-- 1    729088 Aug 19  2009  septerra.exe
    -rw-r--r-- 1    221696 Oct 15  1999  septerra.idx
    -rw-r--r-- 1       594 Oct 15  1999  septerra.ini
    -rw-r--r-- 1       387 Mar 11  2009  septerra.mft
    -rw-r--r-- 1 176828017 Oct 15  1999  sound.db
    -rw-r--r-- 1     62895 Sep 14  2012  Support.ico
    -rw-r--r-- 1    364482 Oct 15  1999  text.db
    -rw-r--r-- 1   1946954 Oct 15  1999  topware.db
    -rw-r--r-- 1    230173 Apr 11 12:30  unins000.dat
    -rw-r--r-- 1   1242944 Apr 11 12:29  unins000.exe
    -rw-r--r-- 1        79 Apr 11 12:30  unins000.ini
    -rw-r--r-- 1     23077 Apr 11 12:30  unins000.msg
    -rw-r--r-- 1   2481622 Oct 15  1999  valkyrie.db

**Comment by TaxAkla** [comment]
```
$ ls -og
total 586228
-rw-rw-r-- 1  74477608 out 15  1999  anim.db
-rwxrwxr-x 1     70656 abr 11 09:56  ddraw.dll
-rw-rw-r-- 1      7507 abr 11 09:53  EULA.txt
-rw-rw-r-- 1     96264 set 14  2012  GameuxInstallHelper.dll
-rw-rw-r-- 1    114629 jul 29  2013  gfw_high.ico
-rw-rw-r-- 1    254784 abr 11 09:57  goggame-1207658742.dll
-rw-rw-r-- 1    114629 abr 11 09:57  goggame-1207658742.ico
-rw-rw-r-- 1       900 abr 11 09:58  goggame-1207658742.info
-rw-rw-r-- 1    255296 jan 24  2014  goggame.dll
-rwxrwxr-x 1      3886 abr 11 09:58  goggame-galaxyFileList.ini
-rw-rw-r-- 1     69248 set 14  2012  gog.ico
-rwxrwxr-x 1      2410 abr 11 09:58  goglog.ini
-rwxrwxr-x 1       744 abr 11 09:58 'Launch Septerra Core.lnk'
-rw-rw-r-- 1 226736914 out 15  1999  level.db
-rwxrwxr-x 1  22706442 abr 11 09:58  M1.avi
-rwxrwxr-x 1         0 abr 11 09:56  m1.db
-rwxrwxr-x 1  14102290 abr 11 09:57  M2.avi
-rwxrwxr-x 1         0 abr 11 09:56  m2.db
-rwxrwxr-x 1   3394410 abr 11 09:57  M3.avi
-rwxrwxr-x 1         0 abr 11 09:56  m3.db
-rwxrwxr-x 1   5699844 abr 11 09:58  M4.avi
-rwxrwxr-x 1         0 abr 11 09:56  m4.db
-rwxrwxr-x 1   6507256 abr 11 09:57  M5.avi
-rwxrwxr-x 1         0 abr 11 09:56  m5.db
-rwxrwxr-x 1   2588792 abr 11 09:57  M6.avi
-rwxrwxr-x 1         0 abr 11 09:56  m6.db
-rwxrwxr-x 1   3095082 abr 11 09:57  M7.avi
-rwxrwxr-x 1         0 abr 11 09:56  m7.db
-rwxrwxr-x 1   7577966 abr 11 09:58  M8.avi
-rwxrwxr-x 1         0 abr 11 09:56  m8.db
-rwxrwxr-x 1  16377146 abr 11 09:58  M9A.avi
-rwxrwxr-x 1         0 abr 11 09:56  m9a.db
-rwxrwxr-x 1  16496670 abr 11 09:58  M9B.avi
-rwxrwxr-x 1         0 abr 11 09:56  m9b.db
-rw-rw-r-- 1   1658551 mar 18  2002  Manual.pdf
-rwxrwxr-x 1   2908912 abr 11 09:57  MONOLITH.avi
-rwxrwxr-x 1         0 abr 11 09:56  monolith.db
-rw-rw-r-- 1   6676480 set 23  2004  QuickTime.qts
-rw-rw-r-- 1     36670 mar 30  2009  README.TXT
drwxrwxr-x 2      4096 abr 11 09:59  savedata
-rwxrwxr-x 1      4972 abr 11 09:59  septerra_d3d9.log
-rw-rw-r-- 1    939075 out 15  1999  septerra.db
-rwxrwxr-x 1        12 abr 11 09:59  septerra.dxvk-cache
-rw-rw-r-- 1    729088 ago 19  2009  septerra.exe
-rw-rw-r-- 1    221696 out 15  1999  septerra.idx
-rwxrwxr-x 1       652 abr 11 09:57  septerra.ini
-rw-rw-r-- 1       387 mar 11  2009  septerra.mft
-rw-rw-r-- 1 176828017 out 15  1999  sound.db
-rw-rw-r-- 1     62895 set 14  2012  Support.ico
-rw-rw-r-- 1    364482 out 15  1999  text.db
-rwxrwxr-x 1   2431586 abr 11 09:58  topware.avi
-rwxrwxr-x 1         0 abr 11 09:56  topware.db
-rw-rw-r-- 1    233317 abr 11 09:53  unins000.dat
-rw-rw-r-- 1   1242944 abr 11 09:53  unins000.exe
-rw-rw-r-- 1        86 abr 11 09:53  unins000.ini
-rw-rw-r-- 1     23077 abr 11 09:53  unins000.msg
-rwxrwxr-x 1    298845 abr 11 09:58  unins001.dat
-rw-rw-r-- 1   1246016 abr 11 09:58  unins001.exe
-rwxrwxr-x 1     23077 abr 11 09:58  unins001.msg
-rwxrwxr-x 1      3545 abr 11 09:57  UOPatchReadme.txt
-rwxrwxr-x 1   2996300 abr 11 09:57  valkyrie.avi
-rwxrwxr-x 1         0 abr 11 09:56  valkyrie.db
-rw-rw-r-- 1    353731 abr 11 09:57  webcache.zip
```

GOG Galaxy has downloaded some files after I've had imported Septerra. 

****
That sdl2 version came from OpenMW PPA. https://launchpad.net/~openmw/+archive/ubuntu/openmw

**Comment by M-HT** [comment]
I see the original movies were replaced with zero length files and .avi versions of the movies were added - maybe using unofficial patch. Can you try the version in the attachment to check movie playback ?

[Septerra-test.tar.gz](https://github.com/M-HT/SR/files/4465986/Septerra-test.tar.gz)

**Comment by TaxAkla** [comment]
![image](https://user-images.githubusercontent.com/40277894/79074553-74e73d80-7cc3-11ea-9fa0-e4ee9618ac5f.png)

Now in the windows installation SR plays the first movie but Black and White and hangs after. It's a progress nonetheless.

**Comment by M-HT** [comment]
So, depending on windows version, the GOG windows installer either installs the original movies or .avi versions of the movies, which means I now have the proper version for testing. Can you try the new version in the attachment to check movie playback ?

[Septerra-test2.tar.gz](https://github.com/M-HT/SR/files/4468728/Septerra-test2.tar.gz)

**Comment by TaxAkla** [comment]
The movies are working now. I will soon start a playthrough to see if I can found any bugs.
I need to ask: these updates will be part to the windows version of SR? As I said I'd love to rename SR and play it with GOG Galaxy.

**Comment by M-HT** [comment]
I released version v1.04.0.4 which fixes the movies and allows setting size of audio buffer - it might help with crackling sounds in wine.
___
And the bug with the game borders - that's how SDL2 works when rendering to higher resolution.

**Comment by TaxAkla** [comment]
Thank you very much. Now even that trick with renaming for GOG Galaxy worked and the crackling sound is gone. Now I have the best of the worlds thanks to you! I'll report any related bug that I found with SR.
